docs/DeveloperGuide.md

```diff
@@ -52,21 +52,69 @@ The `Spending` class inherits from `EntryType` class. It is used to store releva
-password by de-serialising the text at their distinct file paths. Wiagi will then initialise its incomes and spendings
-based on the member in the Storage class.
-![storageLoad.png](./Diagrams/storageLoad.png)
+#### Motivation behind the component:
++ Allows the user to save changes, so that they can resume where they left off.
++ Allows advanced users to edit files directly, enabling fast, manual adjustments.
+
+#### How the Storage Component works:
++ Variables and File Path:
+    + `incomes` → `./incomes.txt`
+    + `spendings` → `./spendings.txt`
+    + `password` → `./password.txt`
+
++ To save edited lists:
+    + It is done when users type `bye`, which signals the end of the program.
+    + The lists are serialised to a user-editable format in their respective files.
+
++ To load saved lists:
+  + It is done upon program startup, when `Wiagi` is constructed.
+  + Within `Wiagi` constructor, it will create a new instance of `Storage`, which will then de-serialise the data at the 
+  `incomes` and `spendings` file paths to an `IncomeList` and `SpendingList` respectively.
+  + `Wiagi` will then retrieve the lists in `Storage` to initialise its lists.
+  + Data corruption in the file triggers an exception, often due to user-editing errors.
+  + For missing files (e.g., new users), files are created and the initialised lists will be empty..
+
++ To load password:
+  + The hashed password will simply be loaded from the password file.
+  + For missing files (e.g., new users), users will be prompted to set a new password at the start of the
+  program. The entered password will then be hashed and stored in a newly created password file.
 
-To save data for current session:
-After the command bye is sent by the user, incomes and spendings will be serialised and overwrite texts in
-their distinct file paths.
-![storageSave.png](./Diagrams/storageSave.png)
+#### Implementation:
+#### Storage class
+The `Storage` class is a class that stores `incomes`, `spendings` and `password`. 
+Upon instantiation, it will call `IncomeListStorage.load()`, `SpendingListStorage.load()` and `LoginStorage.load()`, 
+which will initialise the variables in `Storage` respectively.
+
+#### save method in `IncomeListStorage` `SpendingListStorage`
+Both classes have similar implementation for `save()`, except that `SpendingListStorage` saves budget details on the 
+first line.
++ Format: `daily budget | monthly budget | yearly budget`
++ A for loop will loop through the list, and get each of the attributes of each entry within it and separate them by 
+`|`. Hence, each entry will be written line by line to the file.
++ Format: `amount | description | date | tag | recurrence frequency | last recurrence date | last recurrence day`
+  + E.g. `add income 10 part time /2024-10-10/ *job* ~monthly~` will be stored as
+    `10.0|part time|2024-10-10|job|MONTHLY|2024-10-10|10`
+![storageSave.png](./Diagrams/Storage/saveStorageSequenceDiagram.drawio.png)
```

Comment:
For some reason the diagram are inverted in color, cant see much.

Reply:
same for the other storage diagrams

Reply:
I have updated the diagrams, please let me know if the problem still persists.